executor/set_test.go

```diff
@@ -133,7 +133,8 @@ func (s *testSuite) TestSetVar(c *C) {
 	c.Assert(charset, Equals, "utf8")
 	c.Assert(collation, Equals, "utf8_bin")
 
-	tk.MustExec("set @@character_set_results = NULL")
+	tk.MustExec("set character_set_results = NULL")
+	tk.MustQuery("select @@character_set_results").Check(testkit.Rows(""))
```

Comment:
In MySQL, it's very strange that will return `NULL` instead of `""` in `select @@` but return `""` in `show variable` orz

```
mysql> select @@character_set_results ;
+-------------------------+
| @@character_set_results |
+-------------------------+
| utf8                    |
+-------------------------+
1 row in set (0.00 sec)

mysql> set character_set_results = NULL;
Query OK, 0 rows affected (0.00 sec)

mysql> select @@character_set_results ;
+-------------------------+
| @@character_set_results |
+-------------------------+
| NULL                    |
+-------------------------+
1 row in set (0.00 sec)

mysql> show variables where variable_name='character_set_results';
+-----------------------+-------+
| Variable_name         | Value |
+-----------------------+-------+
| character_set_results |       |
+-----------------------+-------+
1 row in set (0.01 sec)

```

Reply:
I think system variables in MySQL have their own types.